### DISCUSSION TITLE
[General] Fork mainnet to debug and trace issue

### DISCUSSION THREAD

**Discussion by Zohar97** [zkSync-Community-Hub/zksync-developers#574] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

In other EVM networks, I can use Foundry/Hardhat to fork mainnet and trace/debug to see the logic flow from a transaction. How can we do it in zkSync? 
I have try https://docs.zksync.io/build/test-and-debug/foundry but it seems not work, i.e i can't query ERC20 balance by using this.

**Comment by uF4No** [comment]
Use the in-memory-node to fork at a specific block or to replay transactions.

https://docs.zksync.io/build/test-and-debug/in-memory-node#fork-a-network
https://docs.zksync.io/build/test-and-debug/in-memory-node#replay-remote-transactions-locally

**Reply by Zohar97** [reply]
i try with the replay remote tracsactions locally. Returned result was different with actual result of tx. Can you check if anything wrong?
<img width="1448" alt="Screenshot 2024-06-12 at 10 46 41" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/96406007/be594334-f9ae-4d72-b0eb-800fd77ee464">
The tx is success but check with replay tx is failse: https://sepolia-era.zksync.network/tx/0x7119045573862797257e4441ff48bf5a3bc4d133a00d167c18dc955eda12cfac

**Reply by Zohar97** [reply]
similar when i run on mainnet. Check success tx return the failed result

**Reply by Zohar97** [reply]
<img width="1478" alt="Screenshot 2024-06-12 at 10 49 53" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/96406007/15046dae-a858-4594-8f49-5778c45fb15d">
This is example when i run on mainnet tx. Result is failed when tx is success https://era.zksync.network/tx/0x826e5857568934c911f21175b3c189683d6ba50da42069a07e295e7b4f0e9ff0

**Reply by Zohar97** [reply]
@uF4No Can you help to check

**Reply by Zohar97** [reply]
@uF4No can you help to check this

**Reply by uF4No** [reply, marked as answer]
sorry for the late reply @Zohar97 
I was able to replay the testnet transaction but noticed I was using a previous version of era-test-node. I've managed to replicate the issue with the latest version and created the following issue in the repo https://github.com/matter-labs/era-test-node/issues/303

**Reply by Zohar97** [reply]
@uF4No many thanks, please help to check the issue above and let me know if we have any updates
